.pipelines/azure_pipeline_mergedbranches.yaml

```diff
@@ -236,13 +237,170 @@ jobs:
+        docker build --isolation=hyperv --tag ${{ variables.repoImageName }}:$(windowsImageTag)-$(windows2019BaseImageVersion)-unsigned --build-arg WINDOWS_VERSION=$(windows2019BaseImageVersion) --build-arg IMAGE_TAG=$(windowsTelemetryTag) --build-arg WINDOWS_AMA_URL=$(WINDOWS_AMA_URL) .
+
+  - task: PowerShell@2
+    displayName: Extract files to sign
+    inputs:
+      targetType: 'inline'
+      script: |
+        echo "Creating docker container..."
+        docker create --name signingContainer ${{ variables.repoImageName }}:$(windowsImageTag)-$(windows2019BaseImageVersion)-unsigned
+
+        echo "Creating fist party directory..."
+        mkdir -p $(Build.ArtifactStagingDirectory)/fpSigning
+        cd $(Build.ArtifactStagingDirectory)/fpSigning
+
+        echo "Extract CertificateGenerator"
+        docker cp signingContainer:C:\opt\amalogswindows\certgenerator\CertificateGenerator.exe .
+        docker cp signingContainer:C:\opt\amalogswindows\certgenerator\CertificateGenerator.dll .
+
+        echo "Extract LivenessProbe"
+        docker cp signingContainer:C:\opt\amalogswindows\scripts\cmd\livenessprobe.exe .
+
+        echo "Extract ps scripts"
+        mkdir -p powershell
+        docker cp signingContainer:C:\opt\amalogswindows\scripts\powershell .
+
+        echo "Creating OSS directory..."
+        mkdir -p $(Build.ArtifactStagingDirectory)/ossSigning
+        cd $(Build.ArtifactStagingDirectory)/ossSigning
+
+        echo "Extract CertificateGenerator"
+        docker cp signingContainer:C:\opt\amalogswindows\certgenerator\BouncyCastle.Crypto.dll .
+        docker cp signingContainer:C:\opt\amalogswindows\certgenerator\Newtonsoft.Json.dll .
+
+        echo "Extract fluent-bit"
+        mkdir -p fluent-bit
+        docker cp signingContainer:C:\opt\fluent-bit .
+        
+        echo "Extract Ruby"
+        mkdir -p ruby31
+        docker cp signingContainer:C:\ruby31 .
+
+        echo "Extract telegraf"
+        docker cp signingContainer:C:\opt\telegraf\telegraf.exe .
+
+        echo "Extract out oms"
+        docker cp signingContainer:C:\opt\amalogswindows\out_oms.so .
+
+        echo "Removing container..."
+        docker rm signingContainer
+
+        echo "List ArtifactStagingDirectory"
+        ls $(Build.ArtifactStagingDirectory)
+        ls .
+
+  - script: dir $(System.DefaultWorkingDirectory)
+    displayName: 'List files in DefaultWorking Directory'
+
+  - script: dir $(Build.ArtifactStagingDirectory)
+    displayName: 'List files in Staging Directory'
+
+  - task: EsrpCodeSigning@3
+    inputs:
+      ConnectedServiceName: 'AME ESRPContainerInsights Service Connection'
+      FolderPath: '$(Build.ArtifactStagingDirectory)/fpSigning'
+      Pattern: '*.dll,*.exe,*.so,*.ps1'
+      signConfigType: 'inlineSignParams'
```

Comment:
how can we validate if the binaries being signed or not as part of the build?

Reply:
in line 318 to 323, it has validation:                 {
                    "KeyCode" : "CP-230012",
                    "OperationCode" : "SigntoolVerify",
                    "Parameters" : {},
                    "ToolName" : "sign",
                    "ToolVersion" : "1.0"
                }    
also we can see the signed files from logs